src/goLanguageServer.ts

```diff
@@ -47,38 +59,41 @@ let languageServerDisposable: vscode.Disposable;
 let latestConfig: LanguageServerConfig;
 let serverOutputChannel: vscode.OutputChannel;
 
-// startLanguageServer starts the language server (if enabled), returning
-// true on success.
-export async function registerLanguageFeatures(ctx: vscode.ExtensionContext): Promise<boolean> {
-	// Subscribe to notifications for changes to the configuration of the language server.
-	ctx.subscriptions.push(vscode.workspace.onDidChangeConfiguration((e) => watchLanguageServerConfiguration(e)));
-
-	const config = parseLanguageServerConfig();
-	if (!config.enabled) {
-		return false;
+// defaultLanguageProviders is the list of providers currently registered.
+let defaultLanguageProviders: vscode.Disposable[] = [];
+
+// restartCommand is the command used by the user to restart the language
+// server.
+let restartCommand: vscode.Disposable;
+
+// startLanguageServerWithFallback starts the language server, if enabled,
+// or falls back to the default language providers.
+export async function startLanguageServerWithFallback(ctx: vscode.ExtensionContext, activation: boolean) {
+	const cfg = buildLanguageServerConfig();
+	if (cfg.enabled) {
+		// If the language server is gopls, we can check if the user needs to
+		// update their gopls version. We do this only once per VS Code
+		// activation to avoid inundating the user.
+		if (activation && cfg.serverName === 'gopls') {
+			const tool = getTool(cfg.serverName);
+			if (!tool) {
+				return false;
```

Comment:
Did you mean fallback, not 'return a bool'?

Reply:
Yeah, thanks!